src/data_processing/wiki_to_netflix.py

```diff
@@ -0,0 +1,137 @@
+def format_sparql_query(title, year):
+    QUERY = '''
+        SELECT * WHERE {
+            SERVICE wikibase:mwapi {
+                bd:serviceParam wikibase:api "EntitySearch" ;
+                                wikibase:endpoint "www.wikidata.org" ;
+                                mwapi:search "%(Title)s" ;
+                                mwapi:language "en" .
+                ?item wikibase:apiOutputItem mwapi:item .
+            }
+
+            ?item wdt:P31/wdt:P279* wd:Q11424 .
+            
+            {
+                # Get US release date
+                ?item p:P577 ?releaseDateStatement .
+                ?releaseDateStatement ps:P577 ?releaseDate .
+                ?releaseDateStatement pq:P291 wd:Q30 .  
+            }
+            UNION
+            {
+                # Get unspecified release date
+                ?item p:P577 ?releaseDateStatement .
+                ?releaseDateStatement ps:P577 ?releaseDate .
+                FILTER NOT EXISTS { ?releaseDateStatement pq:P291 ?country }
+            }
+        
+            FILTER (YEAR(?releaseDate) = %(Year)d) .
+
+            ?item rdfs:label ?itemLabel .
+            FILTER (lang(?itemLabel) = "en") .
+
+            OPTIONAL {
+                ?item wdt:P136 ?genre .
+                ?genre rdfs:label ?genreLabel .
+                FILTER (lang(?genreLabel) = "en") .
+            }
+
+            OPTIONAL {?item wdt:P57 ?director.
+                            ?director rdfs:label ?directorLabel.
+                            FILTER (lang(?directorLabel) = "en")}
+
+            SERVICE wikibase:label { bd:serviceParam wikibase:language "en" . }
+            }
+    
+        '''
+    return QUERY % {'Title': title, 'Year': year}
+
+# Getting list of movie IDs, genre IDs, and director IDs from request
+def wiki_query(data_csv, user_agent):
+    wiki_movie_ids = []
+    wiki_genres = []
+    wiki_directors = []
+        
+    for row in data_csv:
+        if row[1] is None:
+            continue
+
+        SPARQL = format_sparql_query(row[2], int(row[1]))
+
+        response = requests.post('https://query.wikidata.org/sparql',
+                    headers={'User-Agent': user_agent},
+                    data={
+                    'query': SPARQL,
+                    'format': 'json',
+                    }
+        )
+        response.raise_for_status() 
+        
+        data = response.json()
+        
+        wiki_movie_ids.append(wiki_feature_info(data, 'item'))
+        wiki_genres.append(wiki_feature_info(data, 'genreLabel'))
+        wiki_directors.append(wiki_feature_info(data, 'directorLabel'))
+    
+    return wiki_movie_ids, wiki_genres, wiki_directors
+
+# Calling all functions
+def process_data(test=False):
+    missing_count = 0
+    processed_data = []
+
+    netflix_file = read_netflix_txt(os.path.join(base_dir, 'movie_titles.txt'), test)
+    num_rows = len(netflix_file)
+
+    netflix_csv = os.path.join(base_dir, 'movie_titles.csv')
```

Comment:
It looks like we're still outputting a CSV to the data directory (because `base_dir` is actually `data_dir` and refers to `../../data`, see other comment).

As seen in the `.gitignore` file, let's use a top level `out` directory for the output: `../../out`. You should probably rename `base_dir` while you're at it to avoid confusion.

Reply:
Just updated to have two separate directories for the data folder (data_dir) and out folder (out_dir) for review